core/store/migrations/migration1587580235/migrate.go

```diff
@@ -1,27 +1,21 @@
 package migration1587580235
 
 import (
-	"time"
-
-	"github.com/ethereum/go-ethereum/common"
 	"github.com/jinzhu/gorm"
-	"github.com/pkg/errors"
-	"github.com/smartcontractkit/chainlink/core/store/models"
 )
 
-type logConsumption struct {
-	ID           *models.ID  `gorm:"primary_key"`
-	BlockHash    common.Hash `gorm:"not null;unique_index:idx_unique_log_consumption"`
-	ConsumerType string      `gorm:"not null;unique_index:idx_unique_log_consumption"`
-	ConsumerID   *models.ID  `gorm:"not null;unique_index:idx_unique_log_consumption"`
-	LogIndex     uint        `gorm:"not null;unique_index:idx_unique_log_consumption"`
-	CreatedAt    time.Time
-}
-
-// Migrate adds the LogConsumption table
+// Migrate adds the LogConsumption table and adds a uniqueness constraint on the
+// combination of block_hash / log_index / consumer_id
 func Migrate(tx *gorm.DB) error {
-	if err := tx.AutoMigrate(&logConsumption{}).Error; err != nil {
-		return errors.Wrap(err, "could not add log_consumption table")
-	}
-	return nil
+	return tx.Exec(`
+	CREATE TABLE "log_consumptions" (
+		"id" serial primary key NOT NULL,
+		"block_hash" bytea NOT NULL,
+		"log_index" integer NOT NULL,
+		"consumer_id" uuid REFERENCES job_specs(id) ON DELETE CASCADE NOT NULL,
+		"created_at" timestamp without time zone DEFAULT now() NOT NULL
```

Comment:
Default is not necessary, gorm will do it for you.

I am not entirely sure if default permits expressions actually, so worth checking this doesn't set a fixed time at time of migration and use that for every inserted record into the future.